src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -1348,6 +1345,97 @@ void Terminal::_updateUrlDetection()
     }
 }
 
+// Interns URegularExpression instances so that they can be reused. This method is thread-safe.

```

Comment:
Okay so I thinks I get what this does, but it might deserve a bit of a longer comment than this. 

If I'm not mistaken, this basically creates a process-wide cache of `unique_uregex`s. That way, if someone comes asking for a particular regex, (on any thread), we can hand back the pre-compiled version of it, if we've already seen it. There's also some affordance for flushing out old cache entries, if there's more than 128.

Reply:
TIL i guess. Feel free to ignore me

Reply:
I turned it into a small class so we can more easily extract it in the future. I left it in the same place though because I still feel like it's better to keep this relatively "new" code close to where we're going to add custom patterns for the first time soon.